apps/backoffice-v2/src/pages/Entity/components/DirectorsCallToAction/DirectorsCallToAction.tsx

```diff
@@ -0,0 +1,85 @@
+import { AnyObject } from '@ballerine/ui';
+import { FunctionComponent, useCallback, useMemo, useState } from 'react';
+import { toTitleCase } from 'string-ts';
+import { CallToAction } from '../CallToAction/CallToAction';
+import {
+  ICallToActionDocumentOption,
+  ICallToActionDocumentSelection,
+  ICallToActionProps,
+} from '../CallToAction/interfaces';
+import { getRevisionReasonsForDocument } from './helpers';
+
+interface IDirectorsCallToActionProps extends ICallToActionProps {
+  value: ICallToActionProps['value'] & {
+    props: ICallToActionProps['value']['props'] & {
+      documents: AnyObject[];
+      workflow: AnyObject;
+      onReset?: () => void;
+    };
+  };
+}
+
+export const DirectorsCallToAction: FunctionComponent<IDirectorsCallToActionProps> = ({
+  value,
+  ...restProps
+}) => {
+  const { documents, workflow, onReset, ...restValueProps } = value?.props || {};
+
+  const [documentId, setDocumentId] = useState<string | null>(null);
+
+  const documentsOptions: ICallToActionDocumentOption[] = useMemo(
+    () =>
+      documents.map(({ category, type, id }) => ({
+        name: toTitleCase(`${category as string} - ${type as string}`),
```

Comment:
See instances of `valueOrNA`, otherwise instead of asserting make sure we have a default or at least an empty string over undefined.